src/main/java/com/blamejared/crafttweaker/impl/recipes/handlers/crafttweaker/CTShapedRecipeHandler.java

```diff
@@ -0,0 +1,68 @@
+package com.blamejared.crafttweaker.impl.recipes.handlers.crafttweaker;
+
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.api.recipes.IRecipeHandler;
+import com.blamejared.crafttweaker.api.recipes.IReplacementRule;
+import com.blamejared.crafttweaker.api.recipes.ReplacementHandlerHelper;
+import com.blamejared.crafttweaker.api.util.StringUtils;
+import com.blamejared.crafttweaker.impl.item.MCItemStackMutable;
+import com.blamejared.crafttweaker.impl.recipes.CTRecipeShaped;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.NonNullList;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@IRecipeHandler.For(CTRecipeShaped.class)
+public final class CTShapedRecipeHandler implements IRecipeHandler<CTRecipeShaped> {
+    
+    @Override
+    public String dumpToCommandString(final IRecipeManager manager, final CTRecipeShaped recipe) {
+    
+        final NonNullList<Ingredient> ingredients = recipe.getIngredients();
+        return String.format(
+                "craftingTable.addShaped(%s, %s, %s);",
```

Comment:
Should we print some info if the recipe has a recipeFunction?

Reply:
Added a print for this case. The format is still up for discussion.